site/app/views/GlobalView.php

```diff
@@ -72,6 +72,8 @@ public function header($breadcrumbs, $wrapper_urls, $sidebar_buttons, $notificat
             ];
         }
 
+        // Authorize the default WebSocket pages for the given course, if applicable
+        $this->core->authorizeWebSocketToken();
```

Comment:
Doesn't belong in a view and shouldn't happen every request. One of my other comments should address this though